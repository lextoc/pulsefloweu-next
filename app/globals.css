.divider {
  margin: 3rem 0;
}

.content {
  margin-bottom: 25vh;
}

.cards {
  display: grid;
  grid-template-columns: repeat(auto-fill, minmax(16rem, 20rem));
  gap: 1.5rem;
}
