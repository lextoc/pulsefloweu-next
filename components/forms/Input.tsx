import { Label } from "@/components/forms/Label";

import styles from "./Input.module.css";

export interface IInputProps
  extends React.DetailedHTMLProps<
    React.InputHTMLAttributes<HTMLInputElement>,
    HTMLInputElement
  > {
  label?: string;
}

export default function Input({ label, ...props }: IInputProps) {
  return (
    <>
      {label && <Label>{label}</Label>}
      <input className={styles.root} {...props} />
    </>
  );
}
