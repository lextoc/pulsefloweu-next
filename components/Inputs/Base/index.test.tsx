import { expect } from "@jest/globals";
import { render } from "@testing-library/react";
import React from "react";

import Input, { InputProps } from "./index";

describe("Input component", () => {
  const defaultProps: InputProps = {
    type: "text",
    label: "Input Label",
  };

  it("renders with label", () => {
    const { getByLabelText } = render(<Input {...defaultProps} />);

    const inputElement = getByLabelText("Input Label");

    expect(inputElement).toBeInTheDocument();
  });

  it("applies inverted styles when inverted is true", () => {
    const { container } = render(<Input {...defaultProps} inverted />);

    const inputElement = container.querySelector(".inverted");

    expect(inputElement).toBeInTheDocument();
  });

  it("applies transparent styles when transparent is true", () => {
    const { container } = render(<Input {...defaultProps} transparent />);

    const inputElement = container.querySelector(".transparent");

    expect(inputElement).toBeInTheDocument();
  });

  it("applies small styles when small is true", () => {
    const { container } = render(<Input {...defaultProps} small />);

    const inputElement = container.querySelector(".small");

    expect(inputElement).toBeInTheDocument();
  });

  it("applies fullWidth styles when fullWidth is true", () => {
    const { container } = render(<Input {...defaultProps} fullWidth />);

    const inputElement = container.querySelector(".fullWidth");

    expect(inputElement).toBeInTheDocument();
  });

  it("matches snapshot", () => {
    const { container } = render(<Input {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });
});
