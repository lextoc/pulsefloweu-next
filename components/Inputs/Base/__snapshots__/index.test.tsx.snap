// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input component matches snapshot 1`] = `
<div>
  <div
    class="root "
  >
    <label
      class="root"
      for=":r5:"
    >
      Input Label
    </label>
    <input
      class="input      "
      id=":r5:"
      type="text"
    />
  </div>
</div>
`;
